mikrotik/resource_bgp_peer.go

```diff
@@ -0,0 +1,318 @@
+		return err
+	}
+	if err := d.Set("instance", b.Instance); err != nil {
+		return err
+	}
+	if err := d.Set("keepalive_time", b.KeepAliveTime); err != nil {
+		return err
+	}
+	if err := d.Set("max_prefix_limit", b.MaxPrefixLimit); err != nil {
+		return err
+	}
+	if err := d.Set("max_prefix_restart_time", b.MaxPrefixRestartTime); err != nil {
+		return err
+	}
+	if err := d.Set("multihop", b.Multihop); err != nil {
+		return err
+	}
+	if err := d.Set("nexthop_choice", b.NexthopChoice); err != nil {
+		return err
+	}
+	if err := d.Set("out_filter", b.OutFilter); err != nil {
+		return err
+	}
+	if err := d.Set("passive", b.Passive); err != nil {
+		return err
+	}
+	if err := d.Set("remote_address", b.RemoteAddress); err != nil {
+		return err
+	}
+	if err := d.Set("remote_as", b.RemoteAs); err != nil {
+		return err
+	}
+	if err := d.Set("remote_port", b.RemotePort); err != nil {
+		return err
+	}
+	if err := d.Set("remove_private_as", b.RemovePrivateAs); err != nil {
+		return err
+	}
+	if err := d.Set("route_reflect", b.RouteReflect); err != nil {
+		return err
+	}
+	if err := d.Set("tcp_md5_key", b.TCPMd5Key); err != nil {
+		return err
+	}
+	if err := d.Set("ttl", b.TTL); err != nil {
+		return err
+	}
+	if err := d.Set("update_source", b.UpdateSource); err != nil {
+		return err
+	}
+	if err := d.Set("use_bfd", b.UseBfd); err != nil {
+		return err
+	}
+	return nil
+}
+
+func prepareBgpPeer(d *schema.ResourceData) *client.BgpPeer {
+	bgpPeer := new(client.BgpPeer)
+
+	bgpPeer.Name = d.Get("name").(string)
+	bgpPeer.AddressFamilies = d.Get("address_families").(string)
+	bgpPeer.AllowAsIn = d.Get("allow_as_in").(int)
+	bgpPeer.AsOverride = d.Get("as_override").(bool)
+	bgpPeer.CiscoVplsNlriLenFmt = d.Get("cisco_vpls_nlri_len_fmt").(string)
+	bgpPeer.Comment = d.Get("comment").(string)
+	bgpPeer.DefaultOriginate = d.Get("default_originate").(string)
+	bgpPeer.Disabled = d.Get("disabled").(bool)
+	bgpPeer.HoldTime = d.Get("hold_time").(string)
+	bgpPeer.InFilter = d.Get("in_filter").(string)
+	bgpPeer.Instance = d.Get("instance").(string)
+	bgpPeer.KeepAliveTime = d.Get("keepalive_time").(string)
+	bgpPeer.MaxPrefixLimit = d.Get("max_prefix_limit").(int)
+	bgpPeer.MaxPrefixRestartTime = d.Get("max_prefix_restart_time").(string)
+	bgpPeer.Multihop = d.Get("multihop").(bool)
+	bgpPeer.NexthopChoice = d.Get("nexthop_choice").(string)
+	bgpPeer.OutFilter = d.Get("out_filter").(string)
+	bgpPeer.Passive = d.Get("passive").(bool)
+	bgpPeer.RemoteAddress = d.Get("remote_address").(string)
+	bgpPeer.RemoteAs = d.Get("remote_as").(int)
+	bgpPeer.RemotePort = d.Get("remote_port").(int)
+	bgpPeer.RemovePrivateAs = d.Get("remove_private_as").(bool)
+	bgpPeer.RouteReflect = d.Get("route_reflect").(bool)
+	bgpPeer.TCPMd5Key = d.Get("tcp_md5_key").(string)
+	bgpPeer.TTL = d.Get("ttl").(string)
+	bgpPeer.UpdateSource = d.Get("update_source").(string)
+	bgpPeer.UseBfd = d.Get("use_bfd").(bool)
+
+	return bgpPeer
```

Comment:
Feel free to drop, but I prefer the following. They end up being the same thing, but I like struct literals more than using `new` when possible.
```
return client.BgpPeer{

	Name: d.Get("name").(string),
	AddressFamilies: d.Get("address_families").(string),
	AllowAsIn: d.Get("allow_as_in").(int),

        [...]
        UseBfd: d.Get("use_bfd").(bool),
}
```